app/Controllers/Http/EstablishmentController.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+/* istanbul ignore next */
+// eslint-disable-next-line
+const Establishment = use('App/Models/Establishment')
+
+/* istanbul ignore next */
+class EstablishmentController {
+  async create({ request, response }) {
+    const establishment = await Establishment.create({
+      name: request.input('name'),
+      code: request.input('code'),
+      region_id: request.input('region_id')
+    })
+
+    return response.status(201).json({ establishment })
+  }
+
+  async read({ params, response }) {
+    let results = null
+
+    if (params.id !== undefined) {
+      results = await Establishment.find(params.id)
+    } else {
+      // TODO: Debug, not returning anything
+      results = await Establishment.all()
+    }
+    return response.status(200).json({ results })
+  }
+
+  async update({ request, params, response }) {
+    const establishment = await Establishment.find(params.id)
+
+    if (request.input('name')) {
+      establishment.name = request.input('name')
+    }
+    if (request.input('code')) {
+      establishment.name = request.input('code')
+    }
+    if (request.input('region_id')) {
+      establishment.name = request.input('region_id')
+    }
+
+    await establishment.save()
+    return response.status(200).json({ establishment })
+  }
+
+  async delete({ params, response }) {
+    const establishment = await Establishment.find(params.id)
+
+    await establishment.delete()
+    return response.status(204).json({})
```

Comment:
Should not return "{}", because it is not an empty body and will not return 204 therefore.

Reply:
Resolved by 52d5a27